aws/rust-runtime/aws-http/src/content_encoding.rs

```diff
@@ -276,6 +282,30 @@ where
     len
 }
 
+fn total_rendered_length_of_trailers(header_map: Option<&HeaderMap>) -> u64 {
+    match header_map {
+        Some(header_map) => {
+            let mut total = 0;
+            for key in header_map.keys() {
+                total += key.as_str().len() + ": ".len();
+
+                let mut values = header_map.get_all(key).into_iter();
+
+                if let Some(value) = values.next() {
+                    total += value.len() + CRLF.len();
+                }
+
+                for value in values {
+                    total += value.len() + ",".len();
+                }
```

Comment:
Does this match the actual serialization behavior for multiple values? I thought that hyper would add multiple of the same header name with a different value, but I'm not 100% sure.

Reply:
I'm not sure if it matches what Hyper would do in this situation but it I did test it with S3 and it accepted it. I can test an upload the hyper-style headers if you like.